Update extensions doc (#10778)
PyTorchLightning/pytorch-lightning: docs/source/extensions/loops.rst

----------

> Understanding the default Trainer Loop

Reviewer: ```suggestion
Understanding the Default Trainer Loop
```

.. _override default loops:

Overriding the default Loops
----------------------------

The fastest way to get started with loops, is to override functionality of an existing loop.

> Lightning has 4 main loops it uses: :class:`~pytorch_lightning.loops.fit_loop.FitLoop` for fitting (training and validating),

Reviewer 2: ```suggestion
Lightning has 4 main loops which relies on : :class:`~pytorch_lightning.loops.fit_loop.FitLoop` for fitting (training and validating),
```

    # fit() now uses your fancy loop!
    trainer.fit(...)

> But beware: Loop customization gives you more power and full control over the Trainer and with great power comes great responsibility.

Reviewer 2: Neat !

--------
Here is the full API of methods available in the Loop base class.

> The :class:`~pytorch_lightning.loops.base.Loop` class is the base for all loops in Lighting just like the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base for all models.

Reviewer 3: Might be worth grepping "Lighting" and fixing all typos at once 😂

Author: haha... done!!

> The :class:`~pytorch_lightning.loops.base.Loop` class is the base for all loops in Lightning just like the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base for all models.

Reviewer 2: ```suggestion
The :class:`~pytorch_lightning.loops.base.Loop` class is the base of all loops in the same way as the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base of all models.
```

        torch.nn.Dropout(p=0.1),
        torch.nn.Linear(512, datamodule.num_classes),
    )

>     model = ImageClassifier(backbone="resnet18", head=head, num_classes=datamodule.num_classes, output=Probabilities())

Reviewer 2: Neat !
